core/metrics/bandwidth.go

```diff
@@ -45,21 +45,21 @@ func (bwc *BandwidthCounter) LogRecvMessage(size int64) {
 // Bandwidth is associated with the given protocol.ID and peer.ID.
 func (bwc *BandwidthCounter) LogSentMessageStream(size int64, proto protocol.ID, p peer.ID) {
 	bwc.protocolOut.Get(string(proto)).Mark(uint64(size))
-	bwc.peerOut.Get(string(p)).Mark(uint64(size))
+	bwc.peerOut.Get(string(p.MustMarshalBinary())).Mark(uint64(size))
```

Comment:
Why not `p.String()`?

Reply:
The old code used the bytes. This PR should not change anything semantically. If we used `p.String()` we would be changing semantics.

Reply:
Agreed. Let’s keep this change minimal. We can do a follow up PR if we want to change this interface.